src/effects/VST/VSTEffect.cpp

```diff
@@ -1005,17 +1005,32 @@ namespace
 {
    struct VSTEffectMessage : EffectInstance::Message
    {
-      explicit VSTEffectMessage(VSTEffectSettings settings)
-         : settings{ std::move(settings) }
-      {}
+      typedef std::vector<std::optional<double> > ParamVector;
+
+      // Make a message from a chunk and ID-value pairs
+      explicit VSTEffectMessage(const std::vector<char>& chunk, const ParamVector& params)
+         : mChunk(chunk),
+           mParamsVec(params)
```

Comment:
Then add `move(...)` twice, and this completes the elimination of vector copies

Reply:
right, ok